src/basic/os-util.c

```diff
@@ -1,16 +1,34 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 
 #include "alloc-util.h"
-#include "discover-image.h"
 #include "env-file.h"
 #include "env-util.h"
 #include "fd-util.h"
 #include "fileio.h"
 #include "fs-util.h"
 #include "macro.h"
 #include "os-util.h"
+#include "path-util.h"
 #include "string-util.h"
 #include "strv.h"
+#include "utf8.h"
+
+bool image_name_is_valid(const char *s) {
```

Comment:
why move this over?

Reply:
It was in shared/ before, and it's used in os-util, which is being moved to basic/.